modules/benchmarks/src/main/scala/scalacache/benchmark/OhcBenchmark.scala

```diff
@@ -0,0 +1,64 @@
+package scalacache.benchmark
+
+import java.util.concurrent.TimeUnit
+
+import org.caffinitas.ohc.OHCacheBuilder
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
+
+import scalacache._
+import scalacache.memoization._
+import scalacache.modes.sync._
+import scalacache.ohc.OhcCache
+
+@State(Scope.Thread)
+class OhcBenchmark {
+
+  val underlyingCache =
+    OHCacheBuilder
+      .newBuilder()
+      .keySerializer(OhcCache.stringSerializer)
+      .valueSerializer(OhcCache.stringSerializer)
+      .timeouts(true)
+      .build()
+  implicit val cache: Cache[String] = OhcCache(underlyingCache)
+
+  val key = "key"
+  val value: String = "value"
+
+  def itemCachedNoMemoize(key: String): Id[Option[String]] = {
+    cache.get(key)
+  }
+
+  def itemCachedMemoize(key: String): String = memoizeSync(None) {
+    value
+  }
+
+  // populate the cache
+  cache.put(key)(value)
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  def scalacacheGetNoMemoize(bh: Blackhole) = {
+    bh.consume(itemCachedNoMemoize(key))
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  def scalacacheGetWithMemoize(bh: Blackhole) = {
+    bh.consume(itemCachedMemoize(key))
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  def cache2kGet(bh: Blackhole) = {
+    bh.consume(underlyingCache.get(key))
+  }
+
+  @TearDown
+  def close(): Unit = cache.close()
+
```

Comment:
Did you run the benchmarks? Is OHC actually fast?

Reply:
Yes, I did and the result is this:
```
sbt:benchmarks> jmh:run -i 5 -wi 5 -f1 -t1 .*OhcBenchmark.*
[info] # Run complete. Total time: 00:00:42
[info] Benchmark                              Mode  Cnt    Score    Error  Units
[info] OhcBenchmark.ohcGet                    avgt    5  171.455 ±  1.786  ns/op
[info] OhcBenchmark.scalacacheGetNoMemoize    avgt    5  209.608 ± 16.892  ns/op
[info] OhcBenchmark.scalacacheGetWithMemoize  avgt    5  422.980 ± 11.269  ns/op
```

As a matter of fact, OHC is 3-15 times slower than Caffeine because off-heap cache has serialization/deserialization overhead when getting/setting objects.
But I think OHC can be used to avoid GC pressure when using a very huge number of objects in a very large memory.

Reply:
Yes, I did and the result is this:
```
sbt:benchmarks> jmh:run -i 5 -wi 5 -f1 -t1 .*OhcBenchmark.*
[info] # Run complete. Total time: 00:00:42
[info] Benchmark                              Mode  Cnt    Score    Error  Units
[info] OhcBenchmark.ohcGet                    avgt    5  171.455 ±  1.786  ns/op
[info] OhcBenchmark.scalacacheGetNoMemoize    avgt    5  209.608 ± 16.892  ns/op
[info] OhcBenchmark.scalacacheGetWithMemoize  avgt    5  422.980 ± 11.269  ns/op
```

As a matter of fact, OHC is 3-15 times slower than Caffeine because off-heap cache has serialization/deserialization overhead when getting/setting objects.
But I think OHC can be used to avoid GC pressure when using a very huge number of objects in a very large memory.